HTTP Server.core/RequestRouter.cs

```diff
@@ -22,5 +26,27 @@ public Reply HandleData(Request request, IPathContents pathContents)
 
             return _commandsAndCriteria[_commandsAndCriteria.Count-1].Item2.Execute(request);
         }
+
+        public void AddAction(ICriteria criteria, IHttpHandler httpHandler)
+        {
+            _commandsAndCriteria.Add(Tuple.Create(criteria, httpHandler));
+        }
+
+        private bool IsWrongVersion(Request request)
+        {  
+        return IsValidMethod(request) && request.HttpVersion.Substring(0, 5).Equals("HTTP/") && !request.HttpVersion.Equals("HTTP/1.1");
```

Comment:
but what if i wanted to use this `RequestRouter` class in my server and wanted to support more than just 1.1?  is there a way you can use the pieces that you have now and compose them in a way that you have this version filter in the app layer, rather than changing code in this class?